curriculum/challenges/english/25-front-end-development/lecture-working-with-forms-in-react/67d1a928ae86929a85c1bb6b.md

```diff
@@ -2,13 +2,111 @@
 id: 67d1a928ae86929a85c1bb6b
 title: How Do Forms Work in React?
 challengeType: 11
-videoId: nVAaxZ34khk
+videoId: t9ax7VY00m0
 dashedName: how-do-forms-work-in-react
 ---
 
 # --description--
 
-Watch the video lecture and answer the questions below.
+Watch the video or read the transcript and answer the questions below.
+
+# --transcript--
+
+How do forms work in React?
+
+Forms are fundamental to every web application because they let you handle user input, collect data, and trigger actions.
+
+In React, forms are managed using state or refs, giving you full control over their behavior and validation. These two ways to manage forms are called "controlled"  and "uncontrolled" input.
+
+Let's look at what controlled and uncontrolled inputs are.
+
+Controlled input is the most "React-like" way to handle form inputs. With controlled inputs, you store the input field value in state and update it through `onChange` events. This gives you complete control over the form data and allows instant validation and conditional rendering.
+
+The process works like this: React maintains the form state with the `useState` hook, and you update it on every change. When a user types in an input field, the `onChange` event fires, updates the state, and React re-renders the component with the new value.
+
+```js
+import { useState } from "react";
+
+function App() {
+  const [name, setName] = useState("");
+
+  const handleChange = (e) => {
+    setName(e.target.value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    console.log(name);
+  };
+
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
+        <label htmlFor="name">Your name</label> <br />
+        <input value={name} id="name" onChange={handleChange} type="text" />
+        <button type="submit">Submit</button>
+      </form> 
+    </>
+  );
+}
+
+export default App;
+```
+
+The benefits of controlled inputs include the following:
+
+- Immediate access to the form data.
+
+- You can implement instant validation.
+
+- You can conditionally disable the submit button.
+
+- You can control the input value programmatically.
+
+Uncontrolled inputs on the other hand are seen more in traditional HTML forms. So, instead of handling the inputs through the use`State hook, uncontrolled inputs in HTML maintain their own internal state with the help of the DOM.
```

Comment:
```suggestion
Uncontrolled inputs on the other hand are seen more in traditional HTML forms. So, instead of handling the inputs through the `useState` hook, uncontrolled inputs in HTML maintain their own internal state with the help of the DOM.
```